cli/dcoscli/node/main.py

```diff
@@ -472,9 +472,21 @@ def _list(json_, extra_field_names):
     """
 
     client = mesos.DCOSClient()
+    masters = mesos.MesosDNSClient().hosts('master.mesos.')
+    master_state = client.get_master_state()
     slaves = client.get_state_summary()['slaves']
+    for master in masters:
+        if master['ip'] == master_state['hostname']:
+            master['type'] = 'master (leader)'
+            for key in ('id', 'pid', 'version'):
+                master[key] = master_state.get(key)
```

Comment:
`--field` is supposed to add fields to the table view that are already in the `json` view. Otherwise we're just hardcoding values and it's not clear what options are available to add to the table. I'm happy to add more info to the json view though (and then it will work more naturally for the table view)